internal/metrics/handlers/handler.go

```diff
@@ -0,0 +1,261 @@
+package handlers
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"slices"
+
+	"github.com/go-logr/logr"
+	prometheusv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	prometheusalpha1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1alpha1"
+	"github.com/rhobs/multicluster-observability-addon/internal/addon"
+	"github.com/rhobs/multicluster-observability-addon/internal/metrics/config"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	addonapiv1alpha1 "open-cluster-management.io/api/addon/v1alpha1"
+	clusterv1 "open-cluster-management.io/api/cluster/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	clusterIDLabel = "clusterID"
+)
+
+var (
+	ErrInvalidConfigResourcesCount = errors.New("invalid number of configuration resources")
+	ErrUnsupportedAppName          = errors.New("unsupported app name")
+	ErrMissingImageOverride        = errors.New("missing image override")
+	ErrMissingDesiredConfig        = errors.New("missing desiredConfig in managedClusterAddon.Status.ConfigReferences")
+	ErrMissingProxyConfig          = errors.New("missing envoy proxy config in the prometheus agent configmaps")
+)
+
+type OptionsBuilder struct {
+	Client          client.Client
+	ImagesConfigMap types.NamespacedName
+	RemoteWriteURL  string
+	Logger          logr.Logger
+}
+
+func (o *OptionsBuilder) Build(ctx context.Context, mcAddon *addonapiv1alpha1.ManagedClusterAddOn, managedCluster *clusterv1.ManagedCluster, platform, userWorkloads addon.MetricsOptions) (Options, error) {
+	ret := Options{}
+
+	if !platform.CollectionEnabled && !userWorkloads.CollectionEnabled {
+		return ret, nil
+	}
+
+	ret.ClusterName = managedCluster.Name
+	ret.ClusterID = managedCluster.ObjectMeta.Labels[clusterIDLabel]
+	if ret.ClusterID == "" {
+		ret.ClusterID = managedCluster.Name
+	}
+
+	// Fetch image overrides
+	var err error
+	ret.Images, err = o.getImageOverrides(ctx)
+	if err != nil {
+		return ret, fmt.Errorf("failed to get image overrides: %w", err)
+	}
+
+	// Fetch configuration references
+	configResources, err := o.getAvailableConfigResources(ctx, mcAddon)
+	if err != nil {
+		return ret, fmt.Errorf("failed to get configuration resources: %w", err)
+	}
+
+	// Build Prometheus agents for platform and user workloads
+	if platform.CollectionEnabled {
+		if err := o.buildPrometheusAgent(ctx, &ret, configResources, config.PlatformMetricsCollectorApp); err != nil {
+			return ret, err
+		}
+
+		// Fetch rules and scrape configs
+		ret.Platform.ScrapeConfigs = getResourceByLabelSelector[*prometheusalpha1.ScrapeConfig](configResources, config.PlatformPrometheusMatchLabels)
+		if len(ret.Platform.ScrapeConfigs) == 0 {
+			o.Logger.Info("No scrape configs found for platform metrics")
```

Comment:
WDYT of making these Debug rather than Info? Feels like for normal operations this would be more of a debug log